Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java

```diff
@@ -0,0 +1,215 @@
+package com.SYSC4806_Group13.SYSC4806_Project.Controllers;
+
+import com.SYSC4806_Group13.SYSC4806_Project.Exceptions.MissingAttributeException;
+import com.SYSC4806_Group13.SYSC4806_Project.Exceptions.NotFoundException;
+import com.SYSC4806_Group13.SYSC4806_Project.Model.Listing;
+import com.SYSC4806_Group13.SYSC4806_Project.Model.ListingRepository;
+
+import org.springframework.data.util.Streamable;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+public class ListingRestController {
+
+    private final ListingRepository repo;
+
+    public ListingRestController(ListingRepository repo) {
+        this.repo = repo;
+    }
+
+    @GetMapping("/listings")
+    public List<Listing> getListings(@RequestParam(value = "sellerUserId", required = false) Long sellerUserId) {
+        if (sellerUserId == null) {
+            // Return all listings
+            Iterable<Listing> iterable = repo.findAll();
+            return Streamable.of(iterable).toList();
+        } else {
+            // Return listings matching given sellerUserId
+            return repo.findAllBySellerUserId(sellerUserId);
+        }
+    }
+
+    @GetMapping("/listings/{id}")
+    public Listing getListing(@PathVariable String id) {
```

Comment:
Consider changing the param name to be listingID so that it won't ever get confused with userID or other IDs

Reply:
Updated to listingId